linode/domain/framework_resource.go

```diff
@@ -0,0 +1,230 @@
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error finding the specified Domain",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseComputed(ctx, domain)
+	data.parseNonComputed(ctx, domain)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	client := r.Meta.Client
+	var data DomainModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createOpts := linodego.DomainCreateOptions{
+		Domain:      data.Domain.ValueString(),
+		Type:        linodego.DomainType(data.Type.ValueString()),
+		Group:       data.Group.ValueString(),
+		Description: data.Description.ValueString(),
+		SOAEmail:    data.SOAEmail.ValueString(),
+		RetrySec:    int(data.RetrySec.ValueInt64()),
+		ExpireSec:   int(data.ExpireSec.ValueInt64()),
+		RefreshSec:  int(data.RefreshSec.ValueInt64()),
+		TTLSec:      int(data.TTLSec.ValueInt64()),
+	}
+
+	if !data.MasterIPs.IsNull() && !data.MasterIPs.IsUnknown() {
+		resp.Diagnostics.Append(data.MasterIPs.ElementsAs(ctx, &createOpts.MasterIPs, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	if !data.AXFRIPs.IsNull() && !data.AXFRIPs.IsUnknown() {
+		resp.Diagnostics.Append(data.AXFRIPs.ElementsAs(ctx, &createOpts.AXfrIPs, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	if !data.Tags.IsNull() && !data.Tags.IsUnknown() {
+		resp.Diagnostics.Append(data.Tags.ElementsAs(ctx, &createOpts.Tags, false)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	domain, err := client.CreateDomain(ctx, createOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Domain creation error",
+			err.Error(),
+		)
+		return
+	}
+
+	data.parseComputed(ctx, domain)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var plan, state DomainModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	domainID := int(state.ID.ValueInt64())
+
+	if !domainDeepEqual(plan, state) {
+		r.updateDomain(ctx, resp, plan, domainID)
+	}
+}
+
+func (r *Resource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var data DomainModel
+
+	resp.Diagnostics.Append(resp.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	id := data.ID.ValueInt64()
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+	err := client.DeleteDomain(ctx, int(id))
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); (ok && lerr.Code != 404) || !ok {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Failed to delete domain with id %v", id),
+				err.Error(),
+			)
+		}
+		return
+	}
```

Comment:
May be nicer adding a warning to let the user know that the resource is not found:
```suggestion
	if err != nil {
		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
			resp.Diagnostics.AddWarning(
				"Domain does not exist.",
				fmt.Sprintf("Domain %v does not exist, removing from state.", id),
			)
			return
		} 
		resp.Diagnostics.AddError(
			fmt.Sprintf("Failed to delete domain with id %v", id),
			err.Error(),
		)
		return
	}
```